libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1591,6 +1597,8 @@ bool TypeChecker::visit(TupleExpression const& _tuple)
 			// Note: code generation will visit each of the expression even if they are not assigned from.
 			if (types[i]->category() == Type::Category::RationalNumber && components.size() > 1)
 				if (!dynamic_cast<RationalNumberType const&>(*types[i]).mobileType())
+				// Add Check here (?)
+				// An inadvertent conversion could be happening in RationalType::mobileType()
 					m_errorReporter.fatalTypeError(3390_error, components[i]->location(), "Invalid rational number.");
```

Comment:
I think this one is for stuff like
```solidity
(uint32 a, uint64 b) = (1, 2);
```
or
```solidity
return (1, 2);
```

Currently you cannot use tuple components in expressions so I think there's no point warning about it. You can only assign/pass the whole tuple or use a destructuring assignment.

This will change though if we decide to implement #12635.